net.go

```diff
@@ -118,6 +120,233 @@ func (e *EurekaConnection) GetApps() (map[string]*Application, error) {
+				if pred(instance) {
+					instances = append(instances, instance)
+				}
+			}
+		}
+		return instances
+	}
+}
+
+func (e *EurekaConnection) getInstancesByVIPAddress(addr string, opts instanceQueryOptions) ([]*Instance, error) {
+	var slug string
+	if opts.secure {
+		slug = EurekaURLSlugs["InstancesBySecureVIPAddress"]
+	} else {
+		slug = EurekaURLSlugs["InstancesByVIPAddress"]
+	}
+	reqURL := e.generateURL(slug, addr)
+	log.Debugf("Getting instances for VIP address %q from URL %s", addr, reqURL)
+	body, rcode, err := getBody(reqURL, e.UseJson)
+	if err != nil {
+		return nil, err
+	}
+	if rcode != http.StatusOK {
+		return nil, &unsuccessfulHTTPResponse{rcode, "unable to retrieve instances by VIP address"}
+	}
+	var r *GetAppsResponse
+	if e.UseJson {
+		var rj GetAppsResponseJson
+		err = json.Unmarshal(body, &rj)
+		r = rj.Response
+	} else {
+		err = xml.Unmarshal(body, &r)
+	}
+	if err != nil {
+		log.Errorf("Unmarshalling error: %s", err.Error())
+		return nil, err
+	}
+	var instances []*Instance
+	if pred := opts.predicate; pred != nil {
+		instances = filterInstancesInApps(r.Applications, pred)
+	} else {
+		switch len(r.Applications) {
+		case 0:
+		case 1:
+			instances = r.Applications[0].Instances
+		default:
+			instances = make([]*Instance, instanceCount(r.Applications))
+			base := 0
+			for _, app := range r.Applications {
+				base += copy(instances[base:], app.Instances)
+			}
+		}
+	}
+	if intn := opts.intn; intn != nil {
+		shuffleInstances(instances, intn)
+	}
+	return instances, nil
+}
+
+// GetInstancesByVIPAddress returns the set of instances registered with the given VIP address,
+// potentially filtered per the constraints supplied as options.
+//
+// NB: The VIP address is case-sensitive, and must match the address used at registration time.
+func (e *EurekaConnection) GetInstancesByVIPAddress(addr string, opts ...InstanceQueryOption) ([]*Instance, error) {
+	var mergedOptions instanceQueryOptions
+	for _, o := range opts {
+		if o != nil {
+			if err := o(&mergedOptions); err != nil {
+				return nil, err
+			}
+		}
+	}
+	return e.getInstancesByVIPAddress(addr, mergedOptions)
+}
+
+// GetInstancesBySecureVIPAddress returns the set of instances registered with the given secure
+// VIP address, potentially filtered per the constraints supplied as options.
+//
+// NB: The secure VIP address is case-sensitive, and must match the address used at registration time.
+func (e *EurekaConnection) GetInstancesBySecureVIPAddress(addr string, opts ...InstanceQueryOption) ([]*Instance, error) {
+	mergedOptions := instanceQueryOptions{secure: true}
+	for _, o := range opts {
+		if o != nil {
+			if err := o(&mergedOptions); err != nil {
+				return nil, err
+			}
+		}
+	}
+	return e.getInstancesByVIPAddress(addr, mergedOptions)
+}
+
 // RegisterInstance will register the given Instance with eureka if it is not already registered,
 // but DOES NOT automatically send heartbeats. See HeartBeatInstance for that
```

Comment:
I'm a little bit worried that net.go is getting too big. Do you think this is the case? If so we may want to divide it into smaller pieces. Eg. net/get.go / net/update.go or anything else what makes sense. Feel free to say it's still fine as I don't have a strong opinion here.